github-connector/routing.go

```diff
@@ -8,21 +8,44 @@ import (
 	"github.com/google/go-github/github"
 )
 
-func handleWebhook(w http.ResponseWriter, r *http.Request) {
-	payload, err := github.ValidatePayload(r, []byte("my-secret-key"))
+
+//Validator is an interface used to allow mocking the github library methods
+type Validator interface {
+	ValidatePayload(*http.Request, []byte) ([]byte, error)
+	ParseWebHook(string, []byte) (interface{}, error)
+	GetToken() string
+}
+
+//WebHookHandler is a struct used to allow mocking the github library methods
+type WebHookHandler struct {
+	validator Validator
+}
+
+//NewWebHookHandler creates a new webhook handler with the passed interface
+func NewWebHookHandler(v Validator) *WebHookHandler {
+	return &WebHookHandler{validator: v}
+}
+
+func (wh *WebHookHandler) handleWebhook(w http.ResponseWriter, r *http.Request) {
+
+	payload, err := wh.validator.ValidatePayload(r, []byte(wh.validator.GetToken()))
+
 	if err != nil {
 		log.Printf("error validating request body: err=%s\n", err)
```

Comment:
Does this log provide any info that we need on server side? We have no idea what request did look like, you can either drop it or log the request info along with the error that was met (recommended one, this one may create a lot of logs so think twice and choose a reasonable approach).